app/models/manageiq/providers/nuage/network_manager/vsd_client.rb

```diff
@@ -0,0 +1,159 @@
+      @user = user
+      @password = password
+      @rest_call = Rest.new(server, user, password)
+      _is_conn, data = @rest_call.login
+      if _is_conn
+        @enterprise_id = data
+        return
+      end
+      $log.error('VSD Authentication failed')
+    end
+
+    def get_enterprises
+      response = @rest_call.get(@server + '/enterprises')
+      if response.code == 200
+        if response.body == ''
+          $log.warn('No enterprises present')
+          return
+        end
+        return JSON.parse(response.body)
+      end
+      $log.error('Error in connection ' + response.code.to_s)
+    end
+   
+    def get_domains 
+      response = @rest_call.get(@server + '/domains')
+      if response.code == 200
+        if response.body == ''
+          $log.warn('No domains present')
+          return
+        end
+        return JSON.parse(response.body)
+      end
+      $log.error('Error in connection ' + response.code.to_s)
+    end
+
+    def get_zones
+      response = @rest_call.get(@server + '/zones')
+      if response.code == 200
+        if response.body == ''
+          $log.warn('No zones present')
+          return
+        end
+        return JSON.parse(response.body)
+      end
+      $log.error('Error in connection ' + response.code.to_s)
+    end
+    
+    def get_subnets
+      response = @rest_call.get(@server + '/subnets')
+      if response.code == 200
+        if response.body == ''
+          $log.warn('No subnets present')
+          return
+        end
+        subnets = JSON.parse(response.body)
+        return subnets
+      end
+      $log.error('Error in connection ' + response.code.to_s)
+    end
+
+    def get_vports
+      response = @rest_call.get(@server + '/vports')
+      if response.code == 200
+        if response.body == ''
+          $log.warn('No vports present')
+          return
+        end
+        return JSON.parse(response.body)
+      end
+      $log.error('Error in connection ' + response.code.to_s)
+    end
+
+    def get_vms
+      response = @rest_call.get(@server + '/vms')
+      if response.code == 200
+        if response.body == ''
+          $log.warn('No VM present')
+          return
+        end
+        return JSON.parse(response.body)
+      end
+      $log.error('Error in connection ' + response.code.to_s)
+    end
+
+    def install_license(license_str)
+      license_dict = {}
+      license_dict['license'] = license_str
+      response = @rest_call.post(@server + '/licenses', JSON.dump(license_dict))
+      if response.code != 201
+        if response.body == 'The license already exists'
+          $log.error('license install failed')
+        end
+      end
+    end
+
+    def add_csproot_to_cms_group
+      response = @rest_call.get(@server + "/enterprises/#{@enterprise_id}/groups")
+      groups = JSON.parse(response.body)
+      @cms_group_id = nil
+      csproot_user_id = nil
+
+      groups.each do |group|
+        $log.info('group::' + group)
```

Comment:
for logs like this use interpolation $log.info("group::#{group}") you can also just log all groups with $log.info("groups::#{groups}")